master/buildbot/reporters/gerrit.py

```diff
@@ -401,42 +407,68 @@ def getProperty(build, name):
                     revision = None
 
             if project is not None and revision is not None:
-                self.sendCodeReview(project, revision, result)
+                self.sendCodeReview(project, revision, result, getProperty(build, "event.change.id"))
                 return
 
-    def sendCodeReview(self, project, revision, result):
-        gerrit_version = self.getCachedVersion()
-        if gerrit_version is None:
-            self.callWithVersion(
-                lambda: self.sendCodeReview(project, revision, result))
-            return
+    @defer.inlineCallbacks
+    def sendCodeReview(self, project, revision, result, change_id=None):
+        if self._http == None:
+            gerrit_version = self.getCachedVersion()
+            if gerrit_version is None:
+                self.callWithVersion(
```

Comment:
callWithVersion do not make much sense now that sendCodeReview is inlineCallback. Please remove the function and call the getVersion directly from here.